src/entry-client.js

```diff
@@ -130,12 +130,18 @@ export default function initializeClient(createApp, clientOpts) {
                     .filter(c => !shouldIgnoreRouteUpdate(c, fetchDataArgs))
                     .forEach((c) => {
                         const name = getModuleName(c, to);
-                        const existingModule = find(registeredModules, { name });
+                        const existingModule = get(store, `_modulesNamespaceMap.${name}/`);
                         if (existingModule) {
-                            // We already have this module registered, update the
-                            // index to mark it as recent
                             opts.logger.info('Skipping duplicate Vuex module registration:', name);
-                            existingModule.index = moduleIndex++;
+                            // If the module was registered outside of the routing flow,
+                            // we need to add it to registeredModules.
+                            // Otherwise we will update the index to mark it as recent
+                            const registeredModule = find(registeredModules, { name });
+                            if (!registeredModule) {
+                                registeredModules.push({ name, index: moduleIndex++ });
```

Comment:
Instead of duping this, could we take an approach of conditionally rendering the module below based on `_modulesNamespaceMap`?  If we left this `if` block untouched, would we only need to wrap the `registerModule` call?  Is that functionally equivalent?

```
                            // This module may have been registered outside of the
                            // routing flow, so only register it with Vuex if needed -
                            // but add it to our tracking of registeredModules regardless
                            if (get(store, `_modulesNamespaceMap.${name}/`) == null) {
                                store.registerModule(name, c.vuex.module, {
                                    preserveState: store.state[name] != null,
                                });
                            }
```